### DISCUSSION TITLE
Error- "rush postinstall"

### DISCUSSION THREAD

**Discussion by sanam-bahadur** [yume-chan/ya-webadb#473] [Q&A]
Hi,
I am getting errors while running the code: "rush update"

Executing a maximum of 7 simultaneous processes...

==[ unofficial-adb-book ]=========================================[ 1 of 18 ]==
"unofficial-adb-book" did not define any work.

==[ demo ]========================================================[ 2 of 18 ]==
'fetch-scrcpy-server' is not recognized as an internal or external command,
operable program or batch file.
Returned error code: 1
"demo" failed to build.

==[ @yume-chan/tsconfig ]=========================================[ 3 of 18 ]==
"@yume-chan/tsconfig" did not define any work.

==[ @yume-chan/struct ]===========================================[ 4 of 18 ]==
"@yume-chan/struct" did not define any work.

==[ @yume-chan/stream-extra ]=====================================[ 5 of 18 ]==
"@yume-chan/stream-extra" did not define any work.

==[ @yume-chan/scrcpy-decoder-webcodecs ]=========================[ 6 of 18 ]==
"@yume-chan/scrcpy-decoder-webcodecs" did not define any work.

==[ @yume-chan/scrcpy-decoder-tinyh264 ]==========================[ 7 of 18 ]==
"@yume-chan/scrcpy-decoder-tinyh264" did not define any work.

==[ @yume-chan/scrcpy ]===========================================[ 8 of 18 ]==
"@yume-chan/scrcpy" did not define any work.

==[ @yume-chan/event ]============================================[ 9 of 18 ]==
"@yume-chan/event" did not define any work.

==[ @yume-chan/eslint-config ]===================================[ 10 of 18 ]==
"@yume-chan/eslint-config" did not define any work.

==[ @yume-chan/dataview-bigint-polyfill ]========================[ 11 of 18 ]==
"@yume-chan/dataview-bigint-polyfill" did not define any work.

==[ @yume-chan/b-tree ]==========================================[ 12 of 18 ]==
"@yume-chan/b-tree" did not define any work.

==[ @yume-chan/android-bin ]=====================================[ 13 of 18 ]==
"@yume-chan/android-bin" did not define any work.

==[ @yume-chan/adb-credential-web ]==============================[ 14 of 18 ]==
"@yume-chan/adb-credential-web" did not define any work.

==[ @yume-chan/adb-backend-ws ]==================================[ 15 of 18 ]==
"@yume-chan/adb-backend-ws" did not define any work.

==[ @yume-chan/adb-backend-webusb ]==============================[ 16 of 18 ]==
"@yume-chan/adb-backend-webusb" did not define any work.

==[ @yume-chan/adb-backend-direct-sockets ]======================[ 17 of 18 ]==
"@yume-chan/adb-backend-direct-sockets" did not define any work.

==[ @yume-chan/adb ]=============================================[ 18 of 18 ]==
"@yume-chan/adb" did not define any work.

**Comment by yume-chan** [comment]
I created a new clean Git clone and `rush update` works without issue.

What's your OS and Node version?

**Reply by sanam-bahadur** [reply]
Thanks for the reply. My node version is "16.14.0" and Windows OS. I am trying the same in an organization PC. Is it because of the firewall or proxy?

**Reply by yume-chan** [reply]
I also tried Node.js 16.14.0 (in Docker) and it still works. Does the `rush update` part complete successfully? Can you share the full log?

**Reply by sanam-bahadur** [reply]
Here are the complete logs: 

![P_20230105_104719_1](https://user-images.githubusercontent.com/87244086/210708656-d20ddc42-67a1-4ac9-818b-ae21b1c7b0bd.jpg)
![P_20230105_104617_1](https://user-images.githubusercontent.com/87244086/210708676-f2f96963-298d-47d0-a3ba-1a50d4fb029f.jpg)

**Reply by yume-chan** [reply]
Why are you manually running `rush postinstall`? Did you run `rush update` first?

**Reply by sanam-bahadur** [reply]
yes, we have tried "rush update" first and it was showing the same error.

**Reply by yume-chan** [reply]
What's the full output of `rush update`? Is there a `fetch-scrcpy-server.cmd` in `apps/demo/node_modules/.bin` folder after running `rush update`?

**Reply by sanam-bahadur** [reply]
No, there is nothing in node_modules folder. "fetch-scrcpy-server. cmd" is not created.

Here's the output:
![20230106_124622](https://user-images.githubusercontent.com/87244086/210951121-c77aeb39-305a-4e8e-8e41-6bf2adefd99e.jpg)

**Reply by yume-chan** [reply]
![image](https://user-images.githubusercontent.com/1330321/210952681-2e39aed5-2d89-4ce6-99b2-9da6d978fa00.png)

**Reply by sanam-bahadur** [reply]
Hi, we have deleted the node_modules folder that it is showing and again ran the command "rush update". Then also it is showing same error just not this Warning part.

**Reply by yume-chan** [reply]
Maybe run `git checkout -- .` and `git clean -xdf`, or simply make a new clone, to start all over again.
